File: cadastro-api/src/main/java/com/carlosfilipe/cadastro/service/UserService.java
package com.carlosfilipe.cadastro.service;

import com.carlosfilipe.cadastro.model.User;
import com.carlosfilipe.cadastro.repository.UserRepository;

public class UserService {

    UserRepository repository;

    public User save(User user) {
        repository.save(user);
        return user;
    }
}

File: cadastro-view/src/app/user/user.service.ts
import { Observable, of } from 'rxjs';
import { environment } from '../../environments/environment.prod';
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import { CRUDService } from '../service/CRUDService.service';
import { User } from '../model/user.model';

@Injectable({
    providedIn: 'root'
})
export class UserService extends CRUDService<User> {

  private users: User[] = [];

    constructor(protected http: HttpClient) {
        super(http, environment.cadastroApiUrl, 'usuarios');
    }

    listAll(): Observable<User[]> {
      return of(this.users);
    }
}
